File: src/main/java/miage/skillz/entity/Question.java
package miage.skillz.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name="question")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
//@CheckQuestion
public class Question {
    private static final long serialVersionUID = -2054386655979281969L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long idQuestion;
    private String libelle;
    private String theme;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name="idNiveau")
    private Niveau niveau;

    @Builder.Default
    @OneToMany(mappedBy="question",cascade = CascadeType.ALL)
    private Set<ReponseQuestion> reponsesQuestions = new HashSet<>();

    @JsonIgnore
    @Builder.Default
    @ManyToMany(mappedBy = "quizQuestions",fetch = FetchType.LAZY)
    Set<Quiz> listQuiz = new HashSet<>();

    @Builder.Default
    @ManyToMany(fetch = FetchType.LAZY,cascade = CascadeType.PERSIST)
    @JoinTable( name = "question_competences",
            joinColumns = @JoinColumn(name = "idQuestion"),
            inverseJoinColumns = @JoinColumn(name = "idCompetence"))
    private Set<Competence> questionCompetences = new HashSet<>();

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "idUser")
    private User user;

    public Question(String theme, String libelle,Niveau niveau) {
        this.theme = theme;
        this.libelle = libelle;
        this.niveau = niveau;
    }
}

File: src/main/java/miage/skillz/frontend/src/app/models/question.ts
import { Quiz } from './quiz';
import { ResponseQuestion } from './response-question';
import {Competence} from "./competence";


export class Question {

    idQuestion : number;
    libelle : string;
    theme : string;
    niveau !:  any;
    poids :  number ;
    qst: string;
    reponsesQuestions : Array<ResponseQuestion> = [];

    idNiveau !:  number;
    niveauName!: string;

    questionCompetences : Array<Competence>;

   // listQuizz : Array<Quiz>;

    stringCompetence : string;



    constructor()
    {
        this.idQuestion=0;
        this.theme='';
        this.libelle='';
        this.poids=0;
        this.idNiveau=0;
        this.questionCompetences =[];
      //  this.listQuiz =[];
        this.reponsesQuestions =[];
        this.stringCompetence='';
        this.qst='';


    }
}
